sympy/solvers/solveset.py

```diff
@@ -2256,6 +2441,12 @@ def solvify(f, symbol, domain):
     elif isinstance(solution_set, FiniteSet):
         result = list(solution_set)
 
+    elif isinstance(solution_set, Intersection):
+        result = []
+        for argSet in solution_set.args:
+            if isinstance(argSet,FiniteSet):
+                result += list(argSet)
```

Comment:
here are the modifications in order to give the solution for the `intersection` set...